### PR TITLE
Update scaffold

### PR DESCRIPTION
**Pull request by queil**
This PR is for updating the project structure to Mini-Scaffold to enable fixing #17

Progress:
- [x] src
- [x] test
- [x] ~publish package~ This is already handled by the new scaffold
- [x] docs
- [x] fix source link

### CODE REVIEW

**Comment by queil** [comment]
The Appveyor build fails but the new template comes with GitHub actions ([which work just fine](https://github.com/queil/FsConfig/runs/1674502880?check_suite_focus=true)).

**Comment by queil** [comment]
@tamizhvendan This is ready to go.

**Comment by tamizhvendan** [comment]
Thanks a lot, @queil for the tons of work!

What are all the steps that I need to do post merging this pull request?

**Comment by queil** [comment]
@tamizhvendan I do not think you need to do anything apart from publishing the new version of the nuget package. I've published the package locally and tested #17 - seems to work fine.

**Comment by tamizhvendan** [comment]
@queil Thanks. Let me do it this weekend. Hope it is ok.

**Comment by tamizhvendan** [comment]
Hi @queil It looks like I won't be able to do it this weekend. 

Is it possible for you to do it by yourself?

I've added you as a collaborator to this GitHub project and to add you to the Nuget package management, can you share me your nuget.org's email id.

**Comment by queil** [comment]
@tamizhvendan I've just sent it to your email.

File: paket.dependencies [hunk 1]
```diff
@@ -1,29 +1,62 @@
...
-framework: netstandard2.0, netcoreapp2.0
-redirects: force
-
-nuget FSharp.Core >= 4.2.3
-nuget TypeShape >= 8.0.1
-nuget Microsoft.Extensions.Configuration
+source https://www.nuget.org/api/v2
+source https://api.nuget.org/v3/index.json
+storage: none
+nuget FSharp.Core 5.0
```
Comment (queil) [inline paket.dependencies:4]: @tamizhvendan There is one thing I was not sure about. Which versions of dependencies you want here? The new ones are pretty restrictive. The library still targets `netstandard2.0` though.

Comment (tamizhvendan) [reply paket.dependencies:4]: It looks good to me and I feel we can retain it.

**Comment by tamizhvendan** [comment]
> @tamizhvendan I've just sent it to your email.

@queil Sorry for the delay. I was held up with official work. I have added you as a collaborator for the package. I guess now you have everything to publish the package. 

Feel free to let me know if you need any help

**Pull request merged**